src/redux/epics/index.js

```diff
@@ -62,13 +66,42 @@ const extractNextUrls = (() => {
 
   return ({ link, entry }, depth = 0) => {
     let urls = [extractNextUrlFromLink(link)];
-    if (entry && depth < 4) {
-      urls = urls.concat(entry.map(({ resource }) => extractNextUrls(resource, depth + 1)));
+    if (entry) {
+      if (depth < 2) {
+        urls = urls.concat(entry.map(({ resource }) => extractNextUrls(resource, depth + 1)));
+      } else {
+        console.error('extractReferences depth: ', depth, entry); // eslint-disable-line no-console
+      }
     }
     return flatten(urls).filter((url) => !!url);
   };
 })();
 
+const extractReferences = (() => {
+  const getServiceProvider = (entry) => path(['resource', 'serviceProvider'], entry);
+  const dedupeReferences = (entry) => (entry || [])
+    .map(getServiceProvider)
+    .reduce((acc, serviceProvider) => {
+      // console.info('serviceProvider: ', JSON.stringify(serviceProvider, null, 2));
+      if (serviceProvider?.reference) {
+        acc[serviceProvider.reference] = serviceProvider;
+      }
+      return acc;
+    }, {});
+
+  return ({ entry }, depth = 0) => {
```

Comment:
This is turning out to be a repeating pattern: recursively visiting nested bundles and resources.

TODO: Refactor handling of FHIR responses to 1st flatten this nesting (preserving a reference to appropriate context), which will hopefully enable downstream processing to be simpler.